### PR TITLE
[WIP] Spheroidal polarised dust emission

### PR DESCRIPTION
**Pull request by bwvdnbro**
**Description**
A first implementation of polarised dust emission from spheroidal dust grains, based on tabulated material properties and a model magnetic field. This pull request also contains an additional Probe that can be used to output the magnetic field per grid cell.

**Motivation**
This is the first step in a long project to implement polarised dust emission in SKIRT.

**Tests**
I have created a simple test case containing a cube with a cylindrical hole in the centre that contains a uniform silicon dust component with a single grain size. The box contains a single black-body point source in the centre that heats up the gas and causes (polarised) secondary emission. The entire cylinder is subjected to a cylindrical magnetic field. The test seems to run: linearly polarised light is emitted from the gas in the cylinder. I have currently not been able to check whether this emission is physically correct.

**Guidelines**
I'll leave that as a question. I have attempted to create a `clang-format` style file that can reproduce the SKIRT coding style, but this has proven to be impossible: some of the SKIRT files are inconsistent in terms of formatting (mostly exotic things like single-line function declarations that are present in some files and not in others), some SKIRT style options are not supported by `clang-format` (e.g. line breaks before braces in lambda function declarations), and old SKIRT files are generally also not consistent with `clang-format`s styling for mathematical expressions. `clang-format` is also much better at enforcing the 120 column limit for source code files consistently.
For now, I only applied `clang-format` to the files I changed, and I hope the result is satisfactory. I would be a proponent of using `clang-format` for the entire code base if we can converge to an appropriate style file.

**Context**
This pull request is by no means a finished product and should probably not be merged in just yet (hence the [WIP]). But it would be good to get some feedback about the implementation and the code styling issues.

### CODE REVIEW

File: SKIRT/core/MagneticFieldPerCellProbe.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        // create a text file
+        TextOutFile file(this, itemName() + "_B", "magnetic field per cell");
+
+        // write the header
+        file.writeLine("# Magnetic field per spatial cell");
+        file.addColumn("spatial cell index", "", 'd');
+        file.addColumn("x component of magnetic field", units->umagneticfield(), 'g');
+        file.addColumn("y component of magnetic field", units->umagneticfield(), 'g');
+        file.addColumn("z component of magnetic field", units->umagneticfield(), 'g');
```
Comment (petercamps) [inline SKIRT/core/MagneticFieldPerCellProbe.cpp]: I suggest omitting the 'g' argument at the end. This is what other probes do; it generates nicer columns using the defaults 'e' format.

File: SKIRT/core/MagneticFieldPerCellProbe.hpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+/** MagneticFieldPerCellProbe outputs a column text file (named <tt>prefix_probe_B.dat</tt>)
+    listing the magnetic field for each cell in the spatial grid of the simulation.
+    Specifically, the output file contains a line for each cell in the spatial grid of the
+    simulation. The first column specifies the cell index, and the second, third and fourth
+    column list the magnetic field components. */
+class MagneticFieldPerCellProbe : public Probe
+{
+    ITEM_CONCRETE(MagneticFieldPerCellProbe, Probe, "the magnetic field for each spatial cell")
+        ATTRIBUTE_TYPE_DISPLAYED_IF(MagneticFieldPerCellProbe, "Level2&Dust&SpatialGrid&RadiationField&Panchromatic")
```
Comment (petercamps) [inline SKIRT/core/MagneticFieldPerCellProbe.hpp]: The 'displayed-if' condition is wrong. Refer to another Magnetic Field probe for the correct condition.

Comment (bwvdnbro) [reply SKIRT/core/MagneticFieldPerCellProbe.hpp]: I thought that might be the case. Thanks for the suggestion!

**Comment by bwvdnbro** [comment]
Closed since no longer compatible with the reformatted SKIRT code. Will reintroduce these changes in future pull requests.

**Pull request closed without merging**